### ISSUE TITLE
Rename jax.nn.normalize to jax.nn.standardize

### ISSUE THREAD

**Issue by lucasb-eyer** [google/jax#9899]
As discussed: the jax.nn.normalize function actually standardizes by default, so we should name it accordingly.

Nobody reading `jax.nn.normalize(x)` expects the result not to have norm-1, and [PyTorch agrees with this too](https://pytorch.org/docs/stable/generated/torch.nn.functional.normalize.html).

The proposal here is just to rename to avoid user bugs. This would also open up the name `jax.nn.normalize` for a PyTorch-like implementation, but that should be as a follow-up, if desired.

**Comment by mattjj** [comment]
Hrm, I guess we'd need to do a deprecation cycle for this. The steps may be:
1. in the first PR, rename `jax.nn.normalize` -> `jax.nn.standardize`, but also define `jax.nn.normalize` as an alias;
2. then, add a DeprecationWarning to `jax.nn.normalize` which points to the new API, and in the same CL update all Google-internal users to call `jax.nn.standardize`;
3. make a jax release;
4. wait [three months](https://jax.readthedocs.io/en/latest/api_compatibility.html);
5. either make `jax.nn.normalize` raise an error (instead of a warning) and wait some time, or just delete it;
6. optionally, repurpose `jax.nn.normalize`.

The first two steps could be done in rapid succession.

Anyone want to take it on? 😄

**Comment by dogeplusplus** [comment]
I had a go at implementing this in https://github.com/google/jax/pull/9968. Let me know if I got it wrong or any suggested changes.
